src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -123,12 +123,11 @@ public FeaturesReport SetupFeatures(FeatureConfigurationContext context)
                 }
             }
 
-            settings.PreventChanges();
-
             foreach (var feature in enabledFeatures)
             {
                 ActivateFeature(feature, enabledFeatures, context);
             }
+            settings.PreventChanges();
```

Comment:
Still seems wrong to allow the features to change state when they are activated?